src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -991,6 +991,29 @@
         (autosign-certificate-request! subject csr settings)
         (fs/delete (path-to-cert-request csrdir subject))))))
 
+(schema/defn ^:always-validate process-csr-deletion!
+  :- {:outcome (schema/enum :success :not-found :error) :message schema/Str}
+  "Delete pending certificate requests for subject"
+  [subject :- schema/Str
+   {:keys [csrdir] :as settings} :- CaSettings]
+  (let [csr-path (path-to-cert-request csrdir subject)]
+    (if (fs/exists? csr-path)
+      (if (fs/delete csr-path)
+        (let [msg (str "Deleted certificate request for " subject)]
```

Comment:
Sending a response body and logging seemed like it may be excessive. I could come up with a few diffirent criteria for when to send to a log, a response, or both but I wanted to hear what others thought, and frankly, if people thought that it was worth the effort to do different things for different outcomes.